### PR TITLE
Warn vs error on messages that are too long

### LINKED ISSUE #10: Better error handling for events that are too large
```
ArgumentError: Log Event message cannot be more than than 262144 bytes
```
This error isn't very helpful. It'd be nice if the error printed at least part of the message that it was trying to log.

### PR DESCRIPTION
**Pull request by rofinn**
Closes #10

### CODE REVIEW

**Comment by rofinn** [comment]
Depends on #9 which drops 0.6 and fixes the throttling test.

**Comment by iamed2** [comment]
I think it should not print the truncated log message in the warning. Partly because it's already logging the truncated message, and partly because it could accidentally infinitely recurse.

I do think it should print the full, untruncated message to stderr (or a non-propagating logger which is printing to stderr).

**Comment by rofinn** [comment]
My vote would be to only log the truncated message then. The full message will likely be getting printed stdout via the DefaultHandler anyways.

File: src/event.jl [hunk 1]
```diff
@@ -35,9 +35,11 @@ struct LogEvent
...
 
         if sizeof(message) > MAX_EVENT_SIZE - 26
-            throw(ArgumentError(
-                "Log Event message cannot be more than than $MAX_EVENT_SIZE bytes"
-            ))
+            # Truncate message and throw a warning
+            idx = MAX_EVENT_SIZE - 29
+            # Truncated messages include a "..."
+            message = string(message[1:idx], "...")
+            warn(LOGGER, "Log Event message cannot be more than $MAX_EVENT_SIZE bytes")
```
Comment (iamed2) [inline src/event.jl]:
```suggestion
            warn(LOGGER, "CloudWatch Log Event message cannot be more than $MAX_EVENT_SIZE bytes")
```
Sorry, I squashed this into my previous suggestion but didn't comment on it specifically

**Pull request merged**